src/app/app.component.spec.ts

```diff
@@ -155,4 +158,68 @@ describe('AppComponent', () => {
        flush();
   }));
 
+  // spinner is shown when user is authorized
+
+  it('should hide the spinner, after workspace is (intially) loaded', fakeAsync(() => {
+    resetCalls(mockNg4SpinnerService);
+    when(mockIndexService.reload()).thenReturn(Promise.resolve());
+    when(mockIndexService.refresh()).thenReturn(Promise.resolve([]));
+    when(mockValidationMarkerService.getAllMarkerSummaries(anything())).thenReturn(Promise.resolve([]));
+
+    // when
+    messagingService.publish('workspace.reload.request', null);
+    tick();
+    const [closure] = capture(mockPersistenceService.listFiles).last();
+    closure(null);
+
+    // expect
+    verify(mockNg4SpinnerService.hide()).once();
+    expect().nothing();
+    flush();
+  }));
+
+  it('should still show the spinner, if list files ran into an error', fakeAsync(() => {
+    resetCalls(mockNg4SpinnerService);
+    when(mockIndexService.reload()).thenReturn(Promise.resolve());
+    when(mockIndexService.refresh()).thenReturn(Promise.resolve([]));
+    when(mockValidationMarkerService.getAllMarkerSummaries(anything())).thenReturn(Promise.resolve([]));
+
+    // when
+    messagingService.publish('workspace.reload.request', null);
+    tick();
+    const [, error] = capture(mockPersistenceService.listFiles).last();
+    error(null);
+
+    // expect
+    verify(mockNg4SpinnerService.hide()).never();
+
+    expect().nothing();
+    flush();
+  }));
+
+  it('should make 3 attempts to list files until spinner is hidden!', fakeAsync(() => {
+
+    resetCalls(mockNg4SpinnerService);
+    when(mockIndexService.reload()).thenReturn(Promise.resolve());
+    when(mockIndexService.refresh()).thenReturn(Promise.resolve([]));
+    when(mockValidationMarkerService.getAllMarkerSummaries(anything())).thenReturn(Promise.resolve([]));
+
+    // when
+    messagingService.publish('workspace.reload.request', null);
+    for (let i = 0; i < 4; i++) {
```

Comment:
This breaks when `WORKSPACE_LOAD_RETRY_COUNT` is changed. Not sure if that is a good or a bad thing… :thinking: